### PR TITLE
Implement bgcolor image operation

### LINKED ISSUE #2259: Add filter spec option for background color
Cleaned up photos are often extracted from a background. For example product beauty shots.  It would be nice if these images could be uploaded to wagtail as transparent png files and then templates could specify the background matte to apply to the image for the rendition.  This way the images appear to be transparent but jpeg compression can be used.  Right now one must either use transparent PNG and accept the large file size or apply the jpeg background matte before uploading the image.  If you go with the jpeg option when the page css changes all images on the site have to be replace and reuploaded which is a time consuming and difficult process (also error prone when the images can be displayed on different background colors)

Would it be feasible to add background matte into the filter spec options?  And perhaps rendition format?  So jpeg could be forced using a specific background color?

### PR DESCRIPTION
**Pull request by kaedroho**
Fixes #2259 and #3713

This implements a new filter called ``bgcolor`` which is useful for people who want to force all images to use JPEG format but can't stop their users from uploading PNG files.

Depends on https://github.com/wagtail/Willow/pull/61

### CODE REVIEW

**Comment by kaedroho** [comment]
Willow 1.1 is released now: https://pypi.python.org/pypi/Willow/1.1

**Comment by kaedroho** [comment]
Rebased

File: docs/topics/images.rst [hunk 1]
```diff
@@ -224,6 +224,25 @@ For example, to make the tag always convert the image to a JPEG, use ``format-jp
...
+
+By default, Wagtail will set the background to white. But if a white background
+doesn't fit your design, you can specify a color using the ``bgcolor`` filter.
+
+This filter takes a single argument, which is a CSS 3 or 6 digit hex code
+representing the color you would like to use:
+
+.. code-block:: html+Django
+
+    {# Set the image backgrounds to black #}
```
Comment (tomdyson) [inline docs/topics/images.rst]: backgrounds -> background ?

**Comment by BertrandBordage** [comment]
Merged as 1e332b4b67a7504bb6de4d5c809f3a687c090964.

**Pull request closed without merging**